server/src/test/kotlin/org/itmodreamteam/myrest/server/service/restaurant/RegisterTest.kt

```diff
@@ -29,16 +37,70 @@ class RegisterTest {
     @Autowired
     lateinit var restaurantRepository: RestaurantRepository
 
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    @Autowired
+    lateinit var employeeRepository: EmployeeRepository
+
+    @MockBean
+    lateinit var notificationService: NotificationService
+
     private lateinit var restaurant: Restaurant
+    private lateinit var user: User
 
     @Before
     fun setup() {
+        user = userRepository.save(User("Pizza", "Lover"))
         restaurant = restaurantRepository.save(Restaurant("Pizza", "Italian food", "ИНН"))
     }
 
+    @Test
+    fun `When user register restaurant, user becomes its manager`() {
+        val registeredRestaurant = restaurantService.register(
+            RestaurantRegistrationInfo("Cozy place", "Domestic cuisine", "docs"),
+            user
+        )
+
+        val savedRestaurant = restaurantRepository.findByIdOrNull(registeredRestaurant.id)
+        assertThat(savedRestaurant).isNotNull
+
+        val employee = employeeRepository.findByRestaurant(savedRestaurant!!)
+        assertThat(employee).isNotEmpty
+        assertThat(employee[0].user).isEqualTo(user)
```

Comment:
You are accessing the first element of the collection, but you haven't checked the collection has only one element. Otherwise, why wouldn't we check the other elements?